DevProxy.Abstractions/Plugins/BasePlugin.cs

```diff
@@ -75,7 +75,8 @@ public abstract class BasePlugin<TConfiguration>(
     BasePlugin(logger, urlsToWatch), IPlugin<TConfiguration> where TConfiguration : new()
 {
     private TConfiguration? _configuration;
-    private readonly HttpClient _httpClient = httpClient;
+
+    protected HttpClient PluginHttpClient { get; } = httpClient;

```

Comment:
I'm on the fence if we should do this. On one hand, it's convenient to expose a generic-purpose HttpClient to all plugins since we all have them. On the other hand, I wonder if plugins might need their own HttpClients with specific configuration and it'll get confusing whether they should use the one from the base class or inject their own.

I suggest, we keep the HttpClient in the base class private for now and revisit it in the future if need be.